cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,139 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"path"
+	"strconv"
+	"strings"
+)
+
+func npmExecuteLint(config npmExecuteLintOptions, telemetryData *telemetry.CustomData) {
+	utils := npm.UtilsBundle{}
+
+	err := runNpmExecuteLint(&utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runNpmExecuteLint(utils npm.Utils) error {
+	packageJSONFiles, err := npm.FindPackageJSONFiles(utils)
+	if err != nil {
+		return err
+	}
+
+	packagesWithCilint := findPackagesWithCilint(packageJSONFiles, utils)
+
+	if len(packagesWithCilint) > 0 {
+		err = runCiLint(packagesWithCilint, utils)
+		if err != nil {
+			return err
+		}
+	} else {
+		runDefaultLint(utils)
+	}
+
+	return nil
+}
+
+func findPackagesWithCilint(packageJSONFiles []string, utils npm.Utils) []string {
+	var packagesWithCilint []string
+
+	for _, file := range packageJSONFiles {
+		var packageJSON map[string]interface{}
+
+		packageRaw, err := utils.FileRead(file)
+		if err != nil {
+			log.Entry().Warnf("failed to read %s to check for existence of ci-lint script: %v", file, err)
+			continue
+		}
+
+		_ = json.Unmarshal(packageRaw, &packageJSON)
+
+		scripts, ok := packageJSON["scripts"].(map[string]interface{})
+		if ok {
+			_, ok := scripts["ci-lint"].(string)
+			if ok {
+				packagesWithCilint = append(packagesWithCilint, file)
+				log.Entry().Info("Discovered ci-lint script in " + file)
+			}
+		}
+	}
+	return packagesWithCilint
+}
+
+func runCiLint(packagesWithCilint []string, utils npm.Utils) error {
+	execRunner := utils.GetExecRunner()
+	oldWorkingDirectory, err := utils.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory before executing ci-lint scripts: %w", err)
+	}
+
+	for _, packageJSON := range packagesWithCilint {
+		dir := path.Dir(packageJSON)
+		err := utils.Chdir(dir)
+		if err != nil {
+			return fmt.Errorf("failed to change into directory for executing ci-lint script: %w", err)
+		}
+
+		// Ignore possible errors when invoking ci-lint script to not fail the pipeline based on linting results
+		err = execRunner.RunExecutable("npm", "run", "--silent", "ci-lint")
```

Comment:
Why silient. Shouldn't we fail the pipeline based in linting results?

Reply:
Same as below. Download cache and registry might not be required, but you might not consume features which are centrally implemented in the npm package by directly using the execRunner.

I think this command is quite similar to npmExecuteScripts for all ci-lint without install ... 
Also the npmExecuteScripts could benefit from what you implemented here. It might be got to only run the scripts in these folders where the script is defined as you do it here with findPackagesWithCilint. 
Maybe you can bring both a little bit closer

Reply:
imo we should not do this in the default setting, maybe optional to avoid frustration based on lint rules the developers did not chose.

Reply:
okay. yes. maybe give the option to fail the pipeline because of linting.

Reply:
The silent flag simply suppresses the npm ERR outputs in the logs as they are usually not very helpful. 
I thought we agreed on leaving the behavior as it is in the current Cloud SDK steps, i.e., not fail the pipeline based on linting results?